### ISSUE TITLE
Wrong gradient for `cholesky`

### ISSUE THREAD

**Issue by simsurace** [EnzymeAD/Enzyme.jl#1254]
There seems to be something wrong still with the rules for `cholesky`. The issue here seems distinct from #1250. The following test fails so this case seems not to be covered by the current test suite:
```julia
g(A, b) = sum(cholesky(A * A') \ b)
A = rand(3, 3)
b = rand(3)
test_reverse(g, Active, (A, Duplicated), (b, Duplicated))
```

**Comment by wsmoses** [comment]
If you're able to can you make a version of this with constant inputs, and ideally as little computation in the differentiation as possible (eg no matmul)?

**Comment by simsurace** [comment]
Hmm, it's not easy to get it working with finite differences. If you want, this should also work to show there is an issue:
```
h(A, b) = sum(cholesky(A) \ b)
A = Symmetric(rand(3, 3)) + Matrix{Float64}(I, 3, 3)
b = rand(3)
V = zeros(3, 3); V[1, 1] = 1
dA = Enzyme.make_zero(A)
Enzyme.autodiff(Reverse, h, Duplicated(A, dA), Const(b))
(h(A + 1e-10*V, b) - h(A, b)) / 1e-10 - dA[1, 1] # should be approx zero
```

**Comment by wsmoses** [comment]
Oh I meant more can you say A = [0.3, 0.4, ...].

**Comment by simsurace** [comment]
Sure, if it means something to you:
```julia

h(A, b) = sum(cholesky(A) \ b)
A = [1.3 0.5; 0.5 1.5]
b = [1., 2.]
V = [1.0 0.0; 0.0 0.0]
dA = Enzyme.make_zero(A)
Enzyme.autodiff(Reverse, h, Duplicated(A, dA), Const(b))
(h(A + 1e-10*V, b) - h(A, b)) / 1e-10 - dA[1, 1] # should be approx zero

```

**Comment by simsurace** [comment]
I'm trying to figure out why the rule implemented in #1220 is so different from the one in ChainRules, i.e. much shorter and not doing any real computation:

https://github.com/JuliaDiff/ChainRules.jl/blob/e569283d59d08a65244bed9c37f49fdefbe541ab/src/rulesets/LinearAlgebra/factorization.jl#L491C1-L528C4
